upgrade/upd_2.5.11-to-2.7.0/index.php

```
@@ -235,48 +284,424 @@ public function apply_addsessioncookieprefs()
+    /**
+     * Check if class/libraries/ has already been cleaned up.
+     *
+     * Uses the presence of README.md as the sentinel — it is always present
+     * in a pre-cleanup 2.5.11 installation and never present in 2.7.0.
+     *
+     * @return bool true if already cleaned up (no action needed)
+     */
+    public function check_cleanuplibraries(): bool
+    {
+        return !file_exists(XOOPS_ROOT_PATH . '/class/libraries/README.md');
+    }
```

PHPMD (2.15.0)

[error] The method check_cleanuplibraries is not named in camelCase. (undefined)

(CamelCaseMethodName)